### PR TITLE
Replacing jenkins/ directory with devops/automation/

### PR DESCRIPTION
**Pull request by tj-devel709**
After discussing with @dalexsoto and @mandel-macaque,
we concluded that binding-tools-for-swift was already mostly using Azure Devops and not Jenkins.

This PR does the following:
- Renames jenkins directory to devops/automation/
- Removes the Jenkinsfile and utils.groovy from the now devops/automation/
- Fixes paths to point to the files in devops/automation/
- devops pipelines now looks for azure-pipelines.yml in the devops/automation/ directory

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -10,6 +10,6 @@
 
 * @stephen-hawley
 
-/jenkins @rolfbjarne
+/devops/automation @rolfbjarne
```
Comment (mandel-macaque) [inline CODEOWNERS:13]: @tj-devel709 add your name there too ;)

File: devops/automation/system-dependencies.sh [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 set -o pipefail
 
-cd "$(dirname "$0")/.."
+cd "$(dirname "$0")/../.."
+echo "This is where it is cd $(dirname "$0")/../.."
```
Comment (mandel-macaque) [inline devops/automation/system-dependencies.sh:9]: What is this?

**Review by mandel-macaque — APPROVED** [review]
Small fixes. No blockers.

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

**Review by stephen-hawley — APPROVED** [review]

File: devops/automation/system-dependencies.sh (hunk 2 above)
Comment (tj-devel709) [reply devops/automation/system-dependencies.sh:9]: Oh man good call, I forgot to remove that echo 😮

**Pull request merged**